Client/Frontend/Sync/BraveCore/History/Historyv2.swift

```diff
@@ -68,8 +68,8 @@ class Historyv2: WebsitePresentable {
         }
     }
     
-    public var domain: String? {
-        historyNode.url.domainURL.absoluteString
+    public var domain: Domain? {
+        return Domain.getOrCreate(forUrl: historyNode.url, persistent: true)
```

Comment:
should we persist it when in private mode?

Reply:
You are right I am adding Private mode part, btw do we have to do the same for bookmarks ?

Reply:
no, bookmarks always persist even if you add them in private mode, but history should be gone when you switch back to normal mode

Reply:
Fix at https://github.com/brave/brave-ios/pull/4074/commits/69f9d82a0c96279e34980c676e0334c5569d3007